test/langtools/tools/javac/annotations/typeAnnotations/RichFormatterWithTypeAnnotationsReentrantTest.java

```diff
@@ -0,0 +1,129 @@
+
+    @Test
+    public void test(Path base) throws Exception {
+        Path libClasses = base.resolve("libclasses");
+        Files.createDirectories(libClasses);
+        new JavacTask(tb)
+                .outdir(libClasses)
+                .sources(
+                        """
+                        package lib;
+                        enum Bar {
+                          BAZ
+                        }
+                        """,
+                        """
+                        package lib;
+                        import java.lang.annotation.ElementType;
+                        import java.lang.annotation.Retention;
+                        import java.lang.annotation.RetentionPolicy;
+                        import java.lang.annotation.Target;
+
+                        @Retention(RetentionPolicy.RUNTIME)
+                        @interface Foo {
+                          Bar value();
+                        }
+                        """,
+                        """
+                        package lib;
+                        import java.lang.annotation.ElementType;
+                        import java.lang.annotation.Retention;
+                        import java.lang.annotation.RetentionPolicy;
+                        import java.lang.annotation.Target;
+
+                        @Retention(RetentionPolicy.RUNTIME)
+                        @Target({ElementType.TYPE_USE, ElementType.PARAMETER})
+                        @Foo(Bar.BAZ)
+                        @interface A {}
+                        """,
+                        """
+                        package lib;
+                        public interface M {
+                          String f(@A String k, @A String v);
+                        }
+                        """)
+                .run()
+                .writeAll();
+        Files.delete(libClasses.resolve("lib").resolve("Bar.class"));
+        String code =
+                """
+                import lib.M;
+                class T implements M {
+                }
+                """;
+        List<String> output =
+                new JavacTask(tb)
+                        .classpath(libClasses)
+                        .sources(code)
+                        .run(Task.Expect.FAIL)
+                        .writeAll()
+                        .getOutputLines(Task.OutputKind.DIRECT);
+        // The crash this is a regression test for doesn't reproduce with -XDrawDiagnostics
+        List<String> expected =
+                Arrays.asList(
+                        "warning: unknown enum constant Bar.BAZ",
+                        "  reason: class file for lib.Bar not found",
+                        "T.java:2: error: T is not abstract and does not override abstract method"
+                            + " f(@lib.A String,@lib.A String) in M",
+                        "class T implements M {",
+                        "^",
+                        "1 error",
+                        "1 warning");
```

Comment:
These messages are locale specific. I don't know if we can detect the keys for these messages instead.

Reply:
Thanks! For the purposes of the test the diagnostic text doesn't matter as long as the compiler isn't crashing. I'll look to see if there's a good way to get the diagnostic codes here, another option might be to just relax the assertion to not match on the output.

Reply:
Yep, or running with jtreg options or flags to ensure the locale of the JVM would work too.

Reply:
I updated this to just assert it exits with a regular diagnostic error instead of a crash.